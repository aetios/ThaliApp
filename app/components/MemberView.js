import React from 'react';
import PropTypes from 'prop-types';
import { Image, Text, ViewPropTypes, StyleSheet } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { connect } from 'react-redux';
import { url } from '../url';
import styles from './style/memberView';
import SquareView from './SquareView';

const regex = new RegExp(`^(${url}/media/public)/(avatars/[^\\.]+)\\.(jpg|jpeg|png|gif)`);

const MemberView = (props) => {
  let photo = props.member.photo;
  const matches = regex.exec(photo);

  if (matches) {
    photo = `${matches[1]}/thumbnails/220x220_1/${matches[2]}.${matches[3]}`;
  }

  return (
    <SquareView style={props.style}>
      <Image style={styles.image} source={{ uri: photo }}>
        <LinearGradient colors={['#55000000', '#000000']} style={styles.overlayGradient} />
        <Text style={styles.nameText}>{props.member.name}</Text>
      </Image>
    </SquareView>
  );
};

MemberView.propTypes = {
  member: PropTypes.shape({
    name: PropTypes.string,
    photo: PropTypes.string,
  }).isRequired,
  style: ViewPropTypes.style,
};

const defaultStyles = StyleSheet.create({
});

MemberView.defaultProps = {
  style: defaultStyles,
};

export default connect(() => ({}), () => ({}))(MemberView);
