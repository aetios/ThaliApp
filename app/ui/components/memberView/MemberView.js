import React from 'react';
import PropTypes from 'prop-types';
import { ImageBackground, Text, ViewPropTypes, StyleSheet, TouchableHighlight } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { connect } from 'react-redux';

import styles from './style/MemberView';
import SquareView from './SquareView';
import * as profileActions from '../../../actions/profile';

const MemberView = props => (
  <SquareView style={props.style} size={props.size} >
    <TouchableHighlight
      style={styles.image}
      onPress={() => props.loadProfile(props.token, props.member.pk)}
    >
      <ImageBackground style={styles.image} source={{ uri: props.member.photo }}>
        <LinearGradient colors={['#55000000', '#000000']} style={styles.overlayGradient} />
        <Text style={styles.nameText}>{props.member.name}</Text>
      </ImageBackground>
    </TouchableHighlight>
  </SquareView>
);

MemberView.propTypes = {
  member: PropTypes.shape({
    pk: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    photo: PropTypes.string.isRequired,
  }).isRequired,
  size: PropTypes.number.isRequired,
  style: ViewPropTypes.style,
  token: PropTypes.string.isRequired,
  loadProfile: PropTypes.func.isRequired,
};

const defaultStyles = StyleSheet.create({
});

MemberView.defaultProps = {
  style: defaultStyles,
};

const mapStateToProps = state => ({
  token: state.session.token,
});

const mapDispatchToProps = dispatch => ({
  loadProfile: (token, pk) => dispatch(profileActions.profile(token, pk)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MemberView);
