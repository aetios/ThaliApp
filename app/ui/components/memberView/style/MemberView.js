import { StyleSheet } from 'react-native';
import Colors from '../../../style/Colors';

const styles = StyleSheet.create({
  nameText: {
    fontSize: 11,
    color: Colors.white,
    position: 'absolute',
    left: 6,
    bottom: 4,
    right: 6,
    backgroundColor: Colors.transparent,
  },
  image: {
    flex: 1,
  },
  overlayGradient: {
    position: 'absolute',
    top: '50%',
    bottom: 0,
    left: 0,
    right: 0,
    opacity: 0.6,
  },
});

export default styles;
