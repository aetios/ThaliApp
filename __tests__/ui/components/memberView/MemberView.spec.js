import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import MemberView from '../../../../app/ui/components/memberView/MemberView';
import reducer from '../../../../app/reducers/index';

jest.mock('react-navigation', () => ({
  withNavigation: component => component,
}));

describe('MemberView component', () => {
  const mockStore = configureStore(reducer);
  const initialState = {
    session: {
      token: 'token123',
    },
  };
  const store = mockStore(initialState);

  const member = {
    pk: 1,
    name: 'Lorem ipsum',
    photo: 'http://example.org/example.png',
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemberView store={store} member={member} size={20} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});