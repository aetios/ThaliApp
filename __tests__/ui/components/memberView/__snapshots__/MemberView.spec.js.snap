// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberView component renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "height": 20,
        "width": 20,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://example.org/example.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <BVLinearGradient
        colors={
          Array [
            5570560,
            4278190080,
          ]
        }
        endPoint={undefined}
        locations={null}
        startPoint={undefined}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0.6,
            "position": "absolute",
            "right": 0,
            "top": "50%",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "bottom": 4,
            "color": "#FFFFFF",
            "fontSize": 11,
            "left": 6,
            "position": "absolute",
            "right": 6,
          }
        }
      >
        Lorem ipsum
      </Text>
    </View>
  </View>
</View>
`;
